### ISSUE TITLE
Update for MediatR 12

### ISSUE THREAD

**Issue by jader1313** [fusonic/dotnet-extensions#2]
We are having problems using the Fusonic.Extensions.UnitTests class because it uses MediatR 11 and our classes use MediatR 12. Any plans to update the MediatR version in Fusonic projects?

**Comment by jhartmann123** [comment]
Yes, either that or we somehow get rid of the hard dependency on MediatR.

As MediatR 12 introduces some breaking changes, we're not going to update before v8, which is planned after the .NET8 release in November.

In the meantime you can lock the package version of Mediator to anything before v12, if that's feasable to you:
```
<PackageVersion Include="MediatR" Version="[11.1.0,12.0.0)" />
```
